src/processing/resources/index.js

```diff
@@ -37,8 +37,8 @@ export async function process (ctx, customCharset) {
             var urlReplacer = getResourceUrlReplacer(ctx);
             var processed   = processors[i].processResource(decoded, ctx, charset, urlReplacer);
 
-            if (processed === null)
-                return null;
+            if (processed && processed.err)
+                return await process(ctx, processed.rightCharset);
```

Comment:
Why we need to pass charset and save it to `rightCharset`. AFAIK it's same object what is in the `ctx.contentInfo.charset`, so `process` don't need additional argument. You can use return `restartProcessingSymbol` (https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Symbol) to notify `process` what processing should be restarted.